src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        Storage.removeFruit(transaction.getFruit(), transaction.getQuantity());
+    }

```

Comment:
This class should ensure that the transaction quantity results in a positive balance in the storage. As per the checklist, if the balance is negative after the transaction, a RuntimeException should be thrown.